packages/block-library/src/button/edit.native.js

```diff
@@ -238,18 +248,38 @@ class ButtonEdit extends Component {
 
 	onChangeBorderRadius( newRadius ) {
 		const { setAttributes, attributes } = this.props;
+		const { borderRadiusUnit } = this.state;
 		const { style } = attributes;
 		const newStyle = {
 			...style,
 			border: {
 				...style?.border,
-				radius: newRadius,
+				radius: `${ newRadius }${ borderRadiusUnit }`, // Store the value with the px value so that it works as expected.
```

Comment:
Should this not be specifically `px` but `unit` instead? 
```suggestion
				radius: `${ newRadius }${ borderRadiusUnit }`, // Store the value with the unit so that it works as expected.
```